### DISCUSSION TITLE
Permission permission_denied: The token provided does not match expected scopes.

### DISCUSSION THREAD

**Discussion by aecceyhan** [community/community#57809] [Packages] [answered]
### Select Topic Area

Question

### Body

Hello all,

I am login with npm login --scope=@name_of_org --registry=https://npm.pkg.github.com
I type my user name, token and my email.
Then I see that I successfully logged in.
<img width="980" alt="image" src="https://github.com/community/community/assets/17556123/82df4498-b12b-4967-89b6-bb77aad03700">

npm install @name_of_org/some_name@1.0.24   

I get Permission permission_denied: The token provided does not match expected scopes.

<img width="983" alt="image" src="https://github.com/community/community/assets/17556123/3dffa1ff-4961-4bac-9117-22f74d0f482d">

However my token has to allow this operation fas as I know 

<img width="762" alt="image" src="https://github.com/community/community/assets/17556123/442c97a1-2884-4927-abbc-a8db19b11caa">

<img width="795" alt="image" src="https://github.com/community/community/assets/17556123/e93c5b0a-2055-460b-93b9-5bdcd7d689d9">

Also there is no waiting pat request on organization

<img width="877" alt="image" src="https://github.com/community/community/assets/17556123/dbc6ca0b-6f58-4562-a15a-5ee55b44e6a7">

What I am missing?

**Comment by akshay-sood** [comment]
Did you ever figure this out? I am also having the exact same issue :(

**Comment by cbschuld** [comment]
@aecceyhan I am also curious if you figured this out, the fine grained tokens do not seem to play well with the npm private repo(s).  Curious if anyone accomplished this?

**Comment by GreenPlanet0204** [comment]
@aecceyhan I am also having the exact same issue

**Comment by aecceyhan** [comment, marked as answer]
Hey folks, sorry for the late response. We check the name of the repository and the name on our package JS file is different. So on our case that's why we experience a problem. If this does not fix your issue I am happy to look into the issue together because also I am curious about it.

**Comment by Binary-Ninja-007** [comment]
🔐 Hmm, it seems like you're experiencing an issue with permissions while attempting to install a package using npm. The error message "Permission permission_denied: The token provided does not match expected scopes" indicates that the token you provided does not have the necessary permissions for the requested operation.

Here are a few things you can check to solve this issue:

Token Scopes: Make sure that the token you generated has the required scopes to perform the installation operation. Tokens can have different scopes, and it's possible that the token you used doesn't have the necessary permissions for the specific operation you're trying to perform. Double-check the scopes of your token and ensure it includes the appropriate access.

Organization Settings: If you are installing a package from an organization's GitHub Package Registry, verify that the organization's settings allow user authentication via tokens. Additionally, ensure that the organization's package registry is properly configured to allow installations.

Token Validity: Confirm that the token you're using is valid and hasn't expired. Tokens typically have an expiration date, so it's essential to ensure that you're using an active and valid token. If necessary, generate a new token and try using that instead.

Retry with Registry URL: Instead of using the --registry flag with the npm install command, try setting the package registry in your .npmrc file. Open the file and add the following line: @name_of_org:registry=https://npm.pkg.github.com. Save the file and then retry the installation without the --registry flag.

If none of these solutions resolve the issue, you might need to reach out to GitHub support to get further assistance in troubleshooting the specific problem with your token permissions.

🤞 Good luck, and I hope you'll soon be able to install the package successfully!

**Comment by Varun-Mehrotra** [comment]
I had what seemed like a similar issue, but it just turned out that github packages doesn't support fine grained tokens:
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#about-scopes-and-permissions-for-package-registries

Moving from fine-grained tokens to classic fixed it for me.

**Reply by dannymartinm** [reply]
Happened to me as well, thank you for the comment! It fixed the issue by using a classic token instead.

**Reply by enkechan** [reply]
worked fine for me, thx!

**Reply by kalnode** [reply]
This worked for me too.

**Reply by emmsdan** [reply]
Same here. this worked for me.

**Reply by trinitronx** [reply]
Seems that this is still an issue in Sept. 2025.

For `Packages` access in an Organization, there is an extra setting required:

- Go to: `Organization Settings` -> `Personal Access Tokens (Settings)` -> `Tokens (Classic)`
  - URL:

        https://github.com/organizations/${YOUR_ORG_HERE}/settings/personal-access-tokens?tab=classic

- Turn on setting (radio button): `Allow access via personal access tokens (classic)`
  - **Screenshot:** <img width="1856" height="946" alt="swappy-20250916_093352" src="https://github.com/user-attachments/assets/d9c46d87-758e-4d3b-a637-fc3c81948862" />

Without this setting, and when trying to use a Classic GitHub token, the following error is returned from the `ghcr.io` API:

```json
{
  "errors": [
    {
      "code": "DENIED",
      "message": "permission_denied: `ExampleOrg` forbids access via a personal access token (classic). Please use a GitHub App, OAuth App, or a personal access token with fine-grained permissions."
    }
  ]
}
```

Also, due to the missing fine-grained permissions for `packages:read` and `packages:write`, _**ALL**_ fine-grained tokens will return this error:

```json
{"errors":[{"code":"DENIED","message":"permission_denied: The token provided does not match expected scopes."}]}
```
